react/components/SearchResult.js

```diff
@@ -34,9 +37,12 @@ export default class SearchResult extends Component {
 
   handleLayoutChange = (e, mode) => {
     e.preventDefault()
+    const defaultModes = ['small', 'inline', 'normal']
+    const modeIndex = defaultModes.indexOf(this.state.galleryLayoutMode) !== -1 ? (defaultModes.indexOf(this.state.galleryLayoutMode) + 1) % 3 : 0
```

Comment:
Too long ternary comparison, maybe should be better to split this into a `if-else` statement to improve code legibility.

Reply:
done.